synapse/handlers/room.py

```diff
@@ -1535,20 +1535,13 @@ async def shutdown_room(
             await self.store.block_room(room_id, requester_user_id)
 
         if not await self.store.get_room(room_id):
-            if block:
-                # We allow you to block an unknown room.
-                return {
-                    "kicked_users": [],
-                    "failed_to_kick_users": [],
-                    "local_aliases": [],
-                    "new_room_id": None,
-                }
-            else:
-                # But if you don't want to preventatively block another room,
-                # this function can't do anything useful.
-                raise NotFoundError(
-                    "Cannot shut down room: unknown room id %s" % (room_id,)
-                )
```

Comment:
I'm a bit on the fence about this. I'd think it more helpful to only return a 200 response if something has actually happened. But it's not a hill I'm willing to die on.

Reply:
I don't disagree, but at this point, we don't know if something has actually happened, because we haven't yet tried to delete things. Keeping track of "did anything actually happen" is fiddly, at best.